### PR TITLE
redis driver

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
give the chance  to use redis as 1st citizen db

### Testing Instructions
simply use 
`use Joomla\Database\Redis\RedisDriver;`

added a pr to the cms repo  https://github.com/joomla/joomla-cms/pull/25957 to use this  in /..libraries/src/Cache/Storage/RedisStorage.php

### CODE REVIEW

File: src/Redis/RedisDriver.php [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+/**
+ * Part of the Joomla Framework Database Package
+ *
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Database\Redis;
+
+use Joomla\CMS\Factory;
```
Comment (SharkyKZ) [inline src/Redis/RedisDriver.php]: Can't use CMS classes here.

**Comment by mbabker** [comment]
Not quite sure what the aim is here?  This looks like it's just a factory class to create a `\Redis` instance, it's not really integrating with anything else.

**Comment by mbabker** [comment]
(Also considering this package is geared toward an API for RDBMS, not sure how you would add in drivers for systems like MongoDB or Redis that integrate into the existing API)

**Comment by wilsonge** [comment]
What Michael said - this package is based around SQL packages - having this in here which doesn't implement QueryInterface or DatabaseInterface makes no sense

**Pull request closed without merging**